lxd/ip/qdisc.go

```diff
@@ -0,0 +1,73 @@
+package ip
+
+import (
+	"github.com/lxc/lxd/shared"
+)
+
+// Qdisc represents 'queueing discipline' object
+type Qdisc struct {
+	Dev     string
+	Handle  string
+	Root    bool
+	Ingress bool
+}
+
+func (qdisc *Qdisc) mainCmd() []string {
+	cmd := []string{"qdisc", "add", "dev", qdisc.Dev}
+	if qdisc.Handle != "" {
+		cmd = append(cmd, "handle", qdisc.Handle)
+	}
+	if qdisc.Root == true {
+		cmd = append(cmd, "root")
+	}
+	if qdisc.Ingress == true {
+		cmd = append(cmd, "ingress")
+	}
+	return cmd
+}
+
+// Add adds qdisc to a node
+func (qdisc *Qdisc) Add() error {
+	cmd := qdisc.mainCmd()
+	_, err := shared.RunCommand("tc", cmd...)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Delete deletes qdisc from node
+func (qdisc *Qdisc) Delete() error {
+	cmd := []string{"qdisc", "del", "dev", qdisc.Dev}
+	if qdisc.Root == true {
+		cmd = append(cmd, "root")
+	}
+	if qdisc.Ingress == true {
+		cmd = append(cmd, "ingress")
+	}
+	_, err := shared.RunCommand("tc", cmd...)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// QdiscHtb represents the hierarchy token bucket qdisc object
+type QdiscHtb struct {
```

Comment:
@presztak "HTB" is an acronym for "Hierarchy Token Bucket" and so it should be capitalised in all places used. This includes the type here, so `QdiscHTB`.